src/App.jsx

```diff
@@ -14,70 +15,83 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  const clearComponent = () => {
+    setSelectedGood('');
+  };
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  const addButton = (good) => {
+    setSelectedGood(good);
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {selectedGood
+          ? `${selectedGood} is selected`
+          : 'No goods selected'}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+        {selectedGood && (
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={clearComponent}
+          />
+        )}
+      </h1>
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+      <table className="table">
+        <tbody>
+          {goods.map((good) => {
+            const selected = good === selectedGood;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+            return (
+              <tr
+                data-cy="Good"
+                key={good}
+                className={
+                  selected
+                    ? 'has-background-success-light'
+                    : ''
+                }
+              >
+                <td>
+                  <button
+                    data-cy={
+                      `${selected
+                        ? 'RemoveButton'
+                        : 'AddButton'}`
+                    }
+                    type="button"
+                    className={
+                      `button ${selected
+                        ? 'is-info'
+                        : ''}`
+                    }
+                    onClick={
+                      selected
```

Comment:
Instead of using ternary operator for onClick event, it would be better to keep logic outside of the JSX to keep the code clean and readable.

Reply:
I can't do that because a selection has been made here {goods.map((good) => {
            const selected = good === selectedGood;

            return (
              <tr

Reply:
You can write a function with default data and then call it where you need it with actual data

Reply:
actually i don't know how to make it . I make somethin like this, but it's not working: 
const selectButton= () => {
      if (selected) {
        clearComponent();
      } else {
        selectGood(good);
      }
    };  onClick={selectButton}

Reply:
Feel free to ask for help in the chat

Reply:
I did it, but probably not exactly the way you wanted (I was helped in chat )